src/runtime/functional.ts

```diff
@@ -0,0 +1,8 @@
+export type Producer<T> = () => T;
```

Comment:
I think others may want to bikeshed on this, suggest just putting `Runnable` in `dom-particle.ts` for now since it's the only one in use and not exposed outside it's own file.

Reply:
This is an idea that I mentioned at the summit. 
@shans Do these types appeal to you? Are there areas of the runtime codebase where these could be used immediately or converted gradually? If not, I'll definitely move `Runnable` to the bikeshed and revisit this idea later.

Reply:
Also want to ping @Cypher1

Reply:
I yield to @lindner 's experience here, otherwise these changes look good.

Reply:
I think if you can find a few other examples of where these could be applied then go ahead and check in the file (& follow this CL up with some conversions..)